File: App/Static/scripts/ts/classes/entities/Task.ts
export class Task
{
	private _id: number;
	private _name: string;
	private _description: string;
	private _code: string;
	
	constructor(id: number, name: string, description: string, code: string)
	{
		this._id = id;
		this._name = name;
		this._description = description;
		this._code = code;
	}
	
	get name(): string
	{
		return this._name;
	}
	
	set name(value: string)
	{
		this._name = value;
	}
	
	get description(): string
	{
		return this._description;
	}
	
	set description(value: string)
	{
		this._description = value;
	}
	
	get code(): string
	{
		return this._code;
	}
	
	set code(value: string)
	{
		this._code = value;
	}
}

File: DataAccessLayer/Entities/Task.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using DataAccessLayer.Entities.Constraints.Abstract;

namespace DataAccessLayer.Entities
{
	public class Task
	{
		public Task()
		{
			Labels      = new List<Label>();
			Comments    = new List<Comment>();
			Constraints = new List<TaskConstraintEntityBase>();
		}

		public int Id { get; set; }

		[Required]
		[MaxLength(255)]
		public string Name { get; set; }

		public string Description { get; set; }

		[Required]
		public string Code { get; set; }

		public DateTime CreationDate { get; set; }

		public virtual ICollection<Label>                    Labels      { get; set; }
		public virtual ICollection<Comment>                  Comments    { get; set; }
		public virtual ICollection<TaskConstraintEntityBase> Constraints { get; set; }

		//todo: allow multiple assignees
		public User Assignee   { get; set; }
		public int? AssigneeId { get; set; }

		public virtual User Creator   { get; set; }
		public         int  CreatorId { get; set; }

		public virtual Column Column   { get; set; }
		public         int    ColumnId { get; set; }

		public virtual BacklogTask BacklogTask   { get; set; }
		public         int?        BacklogTaskId { get; set; }

		public virtual History History   { get; set; }
		public         int?    HistoryId { get; set; }
	}
}